### ISSUE TITLE
ColorKeyword and ColorVariable with common practise palletes

### ISSUE THREAD

**Issue by malliapi** [brigade/scss-lint#461]
Given a widely used palette function used with sass:

```
$palettes: (
	base: (
		black: (
			base: #000
		),
		white: (
			base: #fff
		),
		grey: (
			light: #ebebeb,
		),
	),
);

@function palette($color, $tone: 'base', $brand: 'base') {
	@return map-get(map-get(map-get($palettes, $brand), $color), $tone);
}
```

The ColorKeyword and ColorVariable rules fires off whenever a color: palette(white); colour is used. Is this intentional or is it a bug?

**Comment by sds** [comment]
Hey @malliapi,

Did you mean to say `color: palette(base);`, not `white`? (since `white` is a key nested under the `base` palette name?)

There are two separate issues here. One is that we erroneously report a lint for color keys nested within maps (so there's an issue with nesting here). The other issue is that using a color literal `white` as an argument to a function. Please see https://github.com/brigade/scss-lint/issues/175#issuecomment-45644321 for an explanation of why this is expected behavior (ultimately it's the Sass parser that interprets it as a color and `scss-lint` can't tell what your intention is).
